### DISCUSSION TITLE
How can I control when the CNN model is running and visualization?

### DISCUSSION THREAD

**Discussion by noahzn** [bitcraze/discussions#557] [Q&A]
I notice that after flashing the classification CNN, it runs immediately. 
1. If I want to make the drone take off first, and then start to run the CNN model, how should I implement it?
2. Now I need to use the cfclient to see the confidence, and there is no real-time video as shown in this tutorial (https://www.bitcraze.io/documentation/repository/aideck-gap8-examples/master/ai-examples/classification-demo/). How should I check what the camera is seeing when the model is running?

Thank you.

**Comment by hmllr** [comment]
Hi :)
1) You can add a delay to the classification example and wait for something that indicates that the drone took off, for example a message you send from the STM32.
2) There is no real time video shown in the tutorial - there is only a video of the whole scene to illustrate the demo. You could combine the WiFi streamer example with the classification example if you want both the streaming and the CNN.
Hope this helps,
Hanna
